### ISSUE TITLE
`classDefaultCall` doesn't work if callee is not present, but inherited

### ISSUE THREAD

**Issue by taminomara** [EmmyLuaLs/emmylua-analyzer-rust#577]
Type inference for classes that use `classDefaultCall` fails when constructor function is inherited.

For example:

```lua
--- `Foo` defines an `__init` method.
--- @class Foo
local Foo = MakeClass()

--- @param x integer
function Foo:__init(x) end

--- `Bar` inherits `__init` from `Foo`.
--- @class Bar: Foo
local Bar = MakeClass(Foo)

local bar = Bar() -- should complain about missing an argument
--    ^^^ type of `bar` is inferred as `unknown`
```

Relevant `emmylua.rc` part:

```json
"classDefaultCall": {
    "functionName": "__init",
    "forceNonColon": true,
    "forceReturnSelf": true
}
```

**Comment by xuhuanzy** [comment]
At present, constructors are not inherited, nor should they be.

**Comment by xuhuanzy** [comment]
This requires @CppCXY to make the decision.

**Comment by taminomara** [comment]
So, I've investigated several OOP libraries for Lua:

https://github.com/lodsdev/lua-class
https://github.com/Yonaba/Lua-Class-System
https://github.com/kikito/middleclass
https://stevedonovan.github.io/Penlight/api/topics/01-introduction.md.html#Simplifying_Object_Oriented_Programming_in_Lua
https://github.com/jonstoler/class.lua
https://github.com/Yonaba/30log
https://github.com/ForgeMistress/classy/blob/master/profile.lua
https://github.com/rxi/classic
https://github.com/ImagicTheCat/Luaoop
https://github.com/evolbug/lua-clasp

The list includes OOP libraries recommended by Love2d wiki, also I've searched google and reddit. I'm sure there's more, but I feel that 10 is enough.

Literally **all** of them implement constructor inheritance.

Also I'd like to point out that Python and TypeScript inherit constructors, while C++, Java, C# inherit default constructors.

So, I still maintain that constructors should be inherited.

**Comment by taminomara** [comment]
Note: since we want to deprecate `classDefaultCall`, this question is obsolete (see #715).
